scripts/directtools/__init__.py

```diff
@@ -0,0 +1,450 @@
+            cMax = c
+    axes.set_ylim(ymin=0., ymax=cMax / 100.)
+    xMin, xMax = axes.get_xlim()
+    print('Auto E-range: {}...{} meV'.format(xMin, xMax))
+
+
+def _globalnanminmax(workspaces):
+    """Return global minimum and maximum of the given workspaces."""
+    workspaces = _normwslist(workspaces)
+    globalMin = numpy.inf
+    globalMax = -numpy.inf
+    for ws in workspaces:
+        cMin, cMax = nanminmax(ws)
+        if cMin < globalMin:
+            globalMin = cMin
+        if cMax > globalMax:
+            globalMax = cMax
+    return globalMin, globalMax
+
+
+def _label(ws, cut, width, singleWS, singleCut, singleWidth, quantity, units):
+    """Return a line label for a line profile."""
+    ws = _normws(ws)
+    logs = SampleLogs(ws)
+    wsLabel = ''
+    if not singleWS:
+        logs = SampleLogs(ws)
+        T = numpy.mean(logs.sample.temperature)
+        wsLabel = '\\#{:06d} $T$ = {:0.1f} K $E_i$ = {:0.2f} meV'.format(logs.run_number, T, logs.Ei)
+    cutLabel = ''
+    if not singleCut or not singleWidth:
+        cutLabel = quantity + ' = {:0.2f} $\pm$ {:1.2f}'.format(cut, width) + ' ' + units
+    return wsLabel + ' ' + cutLabel
+
+
+def _normws(workspace):
+    """Retrieve workspace from mtd if it is a string, otherwise return as-is."""
+    name = str(workspace)
+    if name:
+        workspace = mtd[name]
+    return workspace
+
+
+def _normwslist(workspaces):
+    """Retrieve workspaces from mtd if they are string, otherwise return as-is."""
+    if not isinstance(workspaces, collections.Iterable) or isinstance(workspaces, str):
+        workspaces = [workspaces]
+    unnormWss = workspaces
+    return [_normws(ws) for ws in unnormWss]
+
+
+def _plottingtime():
+    """Return a string presenting the plotting time."""
+    return time.strftime('%d.%m.%Y %H:%M:%S')
+
+
+def _points(edges):
+    """Return bin centers."""
+    return (edges[:-1] + edges[1:]) / 2
+
+
+def _profiletitle(workspaces, scan, units, cuts, widths, figure):
+    """Add title to line profile figure."""
+    workspaces = _normwslist(workspaces)
+    if not isinstance(cuts, collections.Iterable):
+        cuts = [cuts]
+    if not isinstance(widths, collections.Iterable):
+        widths = [widths] * len(cuts)
+    if len(workspaces) == 1:
+        title = _singledatatitle(workspaces[0])
+    else:
+        logs = SampleLogs(workspaces[0])
+        title = logs.instrument.name + ' ' + _plottingtime()
+    if len(cuts) == 1 and len(widths) == 1:
+        title = title + '\n' + scan + ' = {:0.2f} $\pm$ {:0.2f}'.format(cuts[0], widths[0]) + ' ' + units
+    figure.suptitle(title)
+
+
+def _profileytitle(workspace, axes):
+    """Set the correct y label for profile axes."""
+    if workspace.YUnit() == 'Dynamic susceptibility':
+        axes.set_ylabel("$\\chi''(Q,E)$")
+    else:
+        axes.set_ylabel('$S(Q,E)$')
+
+
+def _removesingularity(ws, epsilon):
+    """Set bins at -epsilon <= x < epsilon  to zero."""
+    Xs = ws.extractX()
```

Comment:
Having `Xs` and `xs` in the same function can be confusing - could at least one of these be renamed something more descriptive?